samples/resourceops/volume_snapshotop_sequential.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import kfp.dsl as dsl
+
+
+@dsl.pipeline(
+    name="VolumeSnapshotOp Sequential",
+    description="The fourth example of the design doc."
+)
+def volume_snapshotop_sequential(url):
+    vop = dsl.VolumeOp(
+        name="create_volume",
+        resource_name="vol1",
+        size="1Gi",
+        modes=dsl.VOLUME_MODE_RWM
+    )
+
+    step1 = dsl.ContainerOp(
+        name="step1_ingest",
+        image="google/cloud-sdk:216.0.0",
+        command=["sh", "-c"],
+        arguments=["mkdir /data/step1 && "
+                   "gsutil cat %s | gzip -c >/data/step1/file1.gz" % url],
+        volumes={"/data": vop.volume}
```

Comment:
How can user mount with subpath here? Can you add an example for it?

Reply:
This is a good comment! The current code does not support it in a super easy way. But the user can always use the generic `.add_volume_mount()` call to specify a specific subPath:

```python
step.add_volume(vop.volume)
custom_volume_mount = V1VolumeMount(
    name=vop.volume.name,
    mount_path="/mnt",
    sub_path="/subpath"
)
step.add_volume_mount(custom_volume_mount)
```

However, we are open to adding this as extra functionality, after this PR is merged. One idea would be to extend the value of the passed `volumes=...` argument to `ContainerOp`s, so that instead of just specifying a volume, we can specify a `(volume, subPath)` tuple, or a `{"volume": volume, "subpath": subpath, "mountPropagation": ..., "readOnly": ... }` dictionary, to expose the full set of options for volume mounts. But this is not the everyday case, and the current approach makes the common case easy and fast.

Reply:
I am fine to have a future PR to capture it. However, subpath seems to be an important use case from my perspective. With subpath, I can arrange step outputs in their own sub-folders and can make sure they are immutable after step completes. Another syntax might be `volume.mount(subpath, readonly=True)`. It think it should be very simple to add but I am fine if you want to add it right after the PR.

Reply:
@hongye-sun Thanks, we will propose a new PR and add it right after this one.